src/main/java/task/Deadline.java

```diff
@@ -0,0 +1,30 @@
+package task;
+
+import java.time.LocalDateTime;
+import java.time.format.DateTimeFormatter;
+
+public class Deadline extends Task {
+    private final DateTimeFormatter formatter = DateTimeFormatter.ofPattern("dd MMM yyyy h:mm a");
+    private final LocalDateTime by;
+
+    /**
+     * Constructs a Deadline object.
+     *
+     * @param description Description of the deadline.
+     * @param by Date and time by which it should be completed.
+     */
+    public Deadline(String description, LocalDateTime by) {
+        super(description);
+        this.by = by;
+    }
+
+    @Override
+    public String toFileString() {
```

Comment:
Same thing as above, can consider writing JavaDoc comment for your overridden methods for more details on the behaviour?